staging/src/k8s.io/client-go/util/jsonpath/parser_test.go

```diff
@@ -61,6 +61,16 @@ var parserTests = []parserTest{
 		newList(), newIdentifier("end"),
 	}, false},
 	{"malformat input", `{\\\}`, []Node{}, true},
+	{"paired parentheses in quotes", `{[?(@.status.nodeInfo.osImage == "Debian GNU/Linux 7 (wheezy)")]}`,
```

Comment:
please simplify these test cases to something minimal for each description, for instance:
`paired parentheses in quotes` tested by `"()"`
otherwise, the parser logic lgtm, so this is probably the last change before I can lgtm this PR